test/oqs_test_endecode.c

```diff
@@ -1,68 +1,101 @@
-
-static int default_libctx = 1;
-
-static OSSL_LIB_CTX *testctx = NULL;
-static OSSL_LIB_CTX *keyctx = NULL;
+static OSSL_LIB_CTX *libctx = NULL;
+static char *modulename = NULL;
+static char *configfile = NULL;
 static char *testpropq = NULL;
+static OSSL_LIB_CTX *keyctx = NULL;
+static OSSL_LIB_CTX *testctx = NULL;
 
-static OSSL_PROVIDER *oqsprov = NULL;
 static OSSL_PROVIDER *dfltprov = NULL;
 static OSSL_PROVIDER *keyprov = NULL;
 
-static EVP_PKEY *oqstest_make_template(const char *type, OSSL_PARAM *genparams)
-{
-    EVP_PKEY *pkey = NULL;
-    EVP_PKEY_CTX *ctx = NULL;
-
-    /*
-     * No real need to check the errors other than for the cascade
-     * effect.  |pkey| will simply remain NULL if something goes wrong.
-     */
-    (void)((ctx = EVP_PKEY_CTX_new_from_name(keyctx, type, testpropq)) != NULL
-           && EVP_PKEY_paramgen_init(ctx) > 0
-           && (genparams == NULL
-               || EVP_PKEY_CTX_set_params(ctx, genparams) > 0)
-           && EVP_PKEY_generate(ctx, &pkey) > 0);
-    EVP_PKEY_CTX_free(ctx);
+#define nelem(a) (sizeof(a)/sizeof((a)[0]))
+
+typedef struct endecode_params_st {
+    char *format;
+    char *structure;
+    char *keytype;
+    char *pass;
+    int selection;
+
+} ENDECODE_PARAMS;
+
+static ENDECODE_PARAMS test_params_list[] = {
+        {"PEM", "PrivateKeyInfo",          NULL, NULL,
+                                                                                         OSSL_KEYMGMT_SELECT_KEYPAIR |
+                                                                                         OSSL_KEYMGMT_SELECT_ALL_PARAMETERS},
+        {"PEM", "EncryptedPrivateKeyInfo", NULL, "Pass the holy handgrenade of antioch", OSSL_KEYMGMT_SELECT_KEYPAIR |
+                                                                                         OSSL_KEYMGMT_SELECT_DOMAIN_PARAMETERS},
+        {"PEM", "SubjectPublicKeyInfo",    NULL, NULL,                                   OSSL_KEYMGMT_SELECT_PUBLIC_KEY |
+                                                                                         OSSL_KEYMGMT_SELECT_ALL_PARAMETERS},
+        {"DER", "PrivateKeyInfo",          NULL, NULL,                                   OSSL_KEYMGMT_SELECT_KEYPAIR |
+                                                                                         OSSL_KEYMGMT_SELECT_ALL_PARAMETERS},
+        {"DER", "EncryptedPrivateKeyInfo", NULL, "Pass the holy handgrenade of antioch", OSSL_KEYMGMT_SELECT_KEYPAIR |
+                                                                                         OSSL_KEYMGMT_SELECT_DOMAIN_PARAMETERS},
+        {"DER", "SubjectPublicKeyInfo",    NULL, NULL,                                   OSSL_KEYMGMT_SELECT_PUBLIC_KEY |
+                                                                                         OSSL_KEYMGMT_SELECT_ALL_PARAMETERS},
+};
+
+static const char *sigalg_names[] = {
+///// OQS_TEMPLATE_FRAGMENT_SIGNATURE_CASES_START
```

Comment:
Is omission of a jinja2 template script intentional? The code in these template brackets would (need to be) generated by a (suitably named "signature_cases.fragment") script in "oqs-template/test/oqs_test_endecode.c". In turn all other fragments in that directory are apparently no longer needed after the re-write. If you'd be willing to dive into this, a correct code change will have been done if "python3 oqs-template/generate.py" created correct code (passing testing, of course) that then also would become part of the PR. If you don't want to get into `jinja2`, no problem, I'll handle then in a followup PR.